Firestore/core/src/firebase/firestore/util/status.cc

```diff
@@ -51,6 +53,14 @@ Status& Status::CausedBy(const Status& cause) {
   absl::StrAppend(&state_->msg, ": ", cause.error_message());
   return *this;
 }
+#endif  // !__APPLE__
+
+Status& Status::WithPlatformError(std::unique_ptr<PlatformError> error) {
+  if (!ok()) {
+    state_->wrapped = std::move(error);
+  }
```

Comment:
Does this make sense?
```c++
Status s = Status::OK();
s.WithPlatformError(a_notnull_platform_error);
```

Or in other words, should we assert that error is nullptr if the status is ok?